Makefile

```diff
@@ -85,30 +131,46 @@ NODEP ?= 1
 endif
 
 #### Files ####
-OBJ_DIR:= build/sa2
+OBJ_DIR  := build/$(PLATFORM)/$(BUILD_NAME)
 ROM      := $(BUILD_NAME).gba
 ELF      := $(ROM:.gba=.elf)
 MAP      := $(ROM:.gba=.map)
-LDSCRIPT := ldscript.txt
 
-C_SUBDIR = src
+ifeq ($(CPU_ARCH),arm)
 ASM_SUBDIR = asm
+ASM_BUILDDIR = $(OBJ_DIR)/$(ASM_SUBDIR)
+
+# no-op
+ASM_PSEUDO_OP_CONV := sed -n 'p'
+else
+# Convert .2byte -> .short and .4byte -> .int
+#  Note that on 32bit architectures .4byte / .int is enough for storing pointers,
+#  but on 64bit targets it would be .8byte / .quad
+#
+# sed expression script by Kurausukun
+ASM_PSEUDO_OP_CONV := sed -i'' -e 's/\.4byte/\.int/g;s/\.2byte/\.short/g'
```

Comment:
Still not confirmed if the `sed -i'' -e` works on linux, it's just something I read on stackoverflow that it should work for both